### PR TITLE
Add headers to signals and queries

### PR DESCRIPTION
**Pull request by Sushisource**
## What was changed
Passes headers to signals and queries appropriately

## Why?
Headers good

## Checklist

1. Closes https://github.com/temporalio/sdk-core/issues/236

3. How was this tested:
Added tests

4. Any docs updates needed?

### CODE REVIEW

File: protos/local/temporal/sdk/core/workflow_activation/workflow_activation.proto [hunk 1]
```diff
@@ -184,6 +184,8 @@ message QueryWorkflow {
     /// The query's function/method/etc name
     string query_type = 2;
     repeated common.Payload arguments = 3;
+    /// Headers attached to the query
+    map<string, common.Payload> header = 5;
```
Comment (bergundy) [inline protos/local/temporal/sdk/core/workflow_activation/workflow_activation.proto:188]: We call this `headers` in `StartWorkflow`, I'd keep that name for consistency.

File: protos/local/temporal/sdk/core/workflow_activation/workflow_activation.proto [hunk 2]
```diff
@@ -192,11 +194,14 @@ message CancelWorkflow {
...
 
-/// Send a signal to a workflow
+// Send a signal to a workflow
 message SignalWorkflow {
     string signal_name = 1;
     repeated common.Payload input = 2;
+    // Identity of the sender of the signal
     string identity = 3;
+    // Headers attached to the signal
+    map<string, common.Payload> header = 5;
```
Comment (bergundy) [inline protos/local/temporal/sdk/core/workflow_activation/workflow_activation.proto:204]: Same

File: protos/local/temporal/sdk/core/workflow_commands/workflow_commands.proto [hunk 3]
```diff
@@ -273,6 +273,8 @@ message SignalExternalWorkflowExecution {
     string signal_name = 4;
     /// Arguments for the handler
     repeated common.Payload args = 5;
+    /// Headers to attach to the signal
+    map<string, common.Payload> header = 6;
```
Comment (bergundy) [inline protos/local/temporal/sdk/core/workflow_commands/workflow_commands.proto:277]: Looks like there's inconsistency here too, `header_fields` and `header`.

File: sdk-core-protos/src/lib.rs [hunk 4]
```diff
@@ -197,8 +197,12 @@ pub mod coresdk {
             T: AsRef<[u8]>,
         {
             fn from(v: T) -> Self {
+                // TODO: Set better encodings, whole data converter deal. Setting anything for now
```
Comment (bergundy) [inline sdk-core-protos/src/lib.rs:200]: I'm assuming this doesn't affect lang

**Review by bergundy — COMMENTED** [review]
Nice!
I'm missing an integration test for queries and had a few minor comments

File: sdk-core-protos/src/lib.rs (hunk 4 above)
Comment (Sushisource) [reply sdk-core-protos/src/lib.rs:200]: It does not

**Comment by Sushisource** [comment]
@bergundy Can't do IT at the moment without adding entire query functionality to prototype SDK, but added to UT stuff that covers necessary path

**Review by bergundy — APPROVED** [review]
LGTM

**Pull request merged**